helix-rest/src/main/java/org/apache/helix/rest/server/resources/metadatastore/MetadataStoreDirectoryAccessor.java

```diff
@@ -220,6 +228,24 @@ public Response getRoutingData() {
     return JSONRepresentation(responseMap);
   }
 
+  @PUT
+  @Path("/routing-data")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response setRoutingData(String jsonContent) {
+    try {
+      Map<String, List<String>> routingData = new ObjectMapper()
```

Comment:
Same thing as above. We can create ObjectMapper once and use it over and over.

Reply:
Although I agree with the same change in `ZkRoutingDataWriter`, in this case it's actually unnecessary because `MetadataStoreDirectoryAccessor` has a per-request cycle. This object mapper will be reconstructed every time anyway. Given that, the object mapper is better a local variable because it's not used anywhere else.

Reply:
On a second thought, the ObjectMapper will be static and is reusable. Making the changes, then.